### PR TITLE
Less global functions, more comments

### PR DESCRIPTION
**Pull request by jwuttke**
This is #84, but without changes to LLData.

### CODE REVIEW

File: Core/Binning/Bin.cpp [hunk 1]
```diff
@@ -15,15 +15,6 @@
 
 #include "Bin.h"
 
-bool BinContains(const Bin1D &bin, double value)
```
Comment (gpospelov) [inline Core/Binning/Bin.cpp]: BinContains is used in SpecularPeak calculations. It is not covered by unit or functional test. Before doing refactoring please provide tests for working code. (If I'm not wrong, m_lower can be bigger than m_upper in some cases).

File: Core/Binning/ConstKBinAxis.cpp [hunk 2]
```diff
@@ -38,37 +37,31 @@ ConstKBinAxis::ConstKBinAxis(const std::string &name, size_t nbins, double start
...
-    bin_boundaries.resize(m_nbins + 1, 0.0);
-    for(size_t i=0; i<bin_boundaries.size(); ++i) {
+    std::vector<double> bin_boundaries(m_nbins + 1, 0.0);
+    for(size_t i=0; i<bin_boundaries.size(); ++i)
         bin_boundaries[i] = std::asin(start_sin + step*i);
-    }
     setBinBoundaries(bin_boundaries);
 }
 
-ConstKBinAxis* ConstKBinAxis::clone() const
```
Comment (gpospelov) [inline Core/Binning/ConstKBinAxis.cpp]: I prefer clone in cpp file. Please consider not to move it unless we all agree on common policy.

File: Core/Binning/ConstKBinAxis.cpp [hunk 3]
```diff
@@ -79,14 +72,13 @@ ConstKBinAxis* ConstKBinAxis::createClippedAxis(double left, double right) const
...
 {
-    if (!IAxis::equals(other)) return false;
-    if (const ConstKBinAxis* otherAxis = dynamic_cast<const ConstKBinAxis*>(&other)) {
-        if (size() != otherAxis->size()) return false;
-        if ( !Numeric::areAlmostEqual(m_start, otherAxis->m_start)) return false;
-        if ( !Numeric::areAlmostEqual(m_end, otherAxis->m_end)) return false;
-        return true;
-    }
-    return false;
+    const ConstKBinAxis* otherAxis = dynamic_cast<const ConstKBinAxis*>(&other);
```
Comment (gpospelov) [inline Core/Binning/ConstKBinAxis.cpp]: Please consider refactoring by providing auxiliary functions for complicated if statements.

File: Core/Binning/IAxis.cpp [hunk 4]
```diff
@@ -31,7 +31,7 @@ size_t IAxis::findIndex(double value) const
     return size();
 }
 
-bool IAxis::equals(const IAxis& other) const
+bool IAxis::sameName(const IAxis& other) const
```
Comment (gpospelov) [inline Core/Binning/IAxis.cpp]: It is wrong refactoring from object oriented point of view. It's pretty common to call parent's method in derived class and then provide additional checks. What if we put more 'state' in IAxis? Please return previous versions of equals back.

File: Core/Instrument/IHistogram.h [hunk 5]
```diff
@@ -101,28 +99,24 @@ class BA_CORE_API_ IHistogram
     //! Returns y-axis bin index for given globalbin (for 2D histograms).
     int getYaxisIndex(size_t i) const;
 
-    //! @brief Returns the value on x-axis corresponding to the global bin index.
+    //! Returns the center of the x-axis bin at the given global bin index.
     //! @param globalbin The global bin index
-    //! @return The center of corresponding bin of the axis
```
Comment (gpospelov) [inline Core/Instrument/IHistogram.h]: So we avoid using of @return doxygen keyword from now? I thought, that in some cases it helps to describe the function meaning more accurate, while leaving details for @return statement.

File: Tests/Functional/PyCore/legacy/intensitydata.py [hunk 6]
```diff
@@ -25,20 +24,14 @@ def test_create_1d_object(self):
...
         data.addAxis("axis0", 10, 0.0, 10.0)
         data.addAxis("axis1", 20, 0.0, 20.0)
         self.assertEqual(200, data.getAllocatedSize())
         self.assertEqual(2, data.getRank())
-        self.assertEqual(0, data.totalSum())
         data.setAllTo(1.0)
-        self.assertEqual(200.0, data.totalSum())
-        for i in range(0, data.getAllocatedSize()):
-            data[i] = data[i]*-1.0
-        self.assertEqual(-200.0, data.totalSum())
```
Comment (gpospelov) [inline Tests/Functional/PyCore/legacy/intensitydata.py]: Please provide functional test for index access iterator (both, read and write) from Python.

**Comment by jwuttke** [comment]
Either reject this in toto, or cherry-pick the commits you like.

**Comment by jwuttke** [comment]
Was meant as a comment. Decide you whether to close.

**Comment by gpospelov** [comment]
Tool some commits into request #101. Closing this one.

**Pull request closed without merging**